doc/release-notes-1.10.rst

```diff
@@ -20,22 +20,39 @@ The following sections provide detailed lists of changes by component.
 Kernel
 ******
 
-* details ...
+* Remove deprecated k_mem_pool_defrag code
+* add memory domain APIs Add the following application-facing memory domain
+  APIs:
+  * k_mem_domain_init() - to initialize a memory domain
+  * k_mem_domain_destroy() - to destroy a memory domain
+  * k_mem_domain_add_partition() - to add a partition into a domain
+  * k_mem_domain_remove_partition() - to remove a partition from a domain
+  * k_mem_domain_add_thread() - to add a thread into a domain
+  * k_mem_domain_remove_thread() - to remove a thread from a domain
+* add k_calloc() which uses kernel heap to implement traditional calloc()
+  semantics.
+* Introduce object validation mechanism: All system calls made from userspace
+  which involve pointers to kernel objects (including device drivers) will need
+  to have those pointers validated; userspace should never be able to crash the
```

Comment:
"should" would be better stated as "must" here (as in, "must never be able to..."